13_biblioteki/13_8_AlgorithmsBLAS.tex

```diff
@@ -0,0 +1,257 @@
+		   $\uparrow$ & \\
+		   S, D, C, Z &  \\
+	\end{tabular} \\ 
+	we: $ Ay = x $, wy: $x \leftarrow y$ \\
+	\textit{Solve Triangular System}: $ x \leftarrow A^{-1}x $\\ 
+	$ A^{-1} = (A^T)^{-1}, A^{-H} = (A^H)^{-1} $ 
+	\begin{tabular}{c c}
+	$
+	\left .
+ 		 \begin{tabular}{c}
+			 $ x \leftarrow A^{-T} x $ \\
+			 $ x \leftarrow A^{-H} x $ 
+  		\end{tabular}
+	\right \}
+	$
+	 &
+		\begin{tabular}{c}
+			$ A \leftarrow A^T $ \\
+			$ A \leftarrow A^H $
+  		\end{tabular} 
+	\end{tabular} \\
+	
+	\textbf{A} - nxn upper or lower - triangular matrix A \\
+	\textbf{uplo} - upper/lower triangular opt \\
+	= 'L' - solve lower triangular system (forward elimin) \\
+	= 'U' - solve upper triangular system (back substitutuion) \\
+	trans - transposition option for A: \\
+	$N - : x \leftarrow A^{-1} x;$ \\
+	$T - : x \leftarrow A^{-T} x;$ \\
+	$C - : x \leftarrow A^{-H} x$ \\
+	\textbf{diag} - whether A is unit triangular i.e. $a_{ii} = 1$ \\
+	N - diag of A stored in array; \\
+	U - diag = 1, \\
+	\textbf{n} - no of rows and columns in A \\
+	\textbf{A} - array containing n-by-n triangular matrix \\
+	\textbf{lda} - the leading dimension of array a (decl. in call p) $lda \geq max(n, 1)$
+	\textbf{x} - array of $lenx = (n - 1) * |incx| + 1 \leftarrow$ rhs $n$-vector $x$ \\
+	\textbf{incx} - increment for x
+	OUTPUT:  $x$ - the solution vector replaces the input \\
+	Notes: subprogram do not check singularity of A \\
+	\end{frame}
+	\begin{frame}[fragile]{Realizacja z użyciem BLAS level 2 III}
+		\textbf{Program ma postać:}
+		\begin{lstlisting}
+		DO 10 J=1, N
+			CALL STRSU('Upper', 'Transpase', 
+			'Non-Unit', J-1, A, LDA, A(1, J), 1)
+		S - A(J, J) - SDOT(J-1, A(1, J), 1, A(1, J), 1)
+		IF (S .LE. ZERO) GO TO 20
+		A(J, J) = SQRT(S)
+		10 CONTINUE
+		\end{lstlisting}
+		
+	\end{frame}
+	\begin{frame}{Realizacja z użyciem BLAS level 2}
+	Improvement of the performance: \\ 
+	1 processor CRAY Y-MP, N= 500 \\
+	\begin{itemize}
+		\item dot version: 72 MFLOPS
+		\item 251 MFLOPS (81\% of A*B)
+	\end{itemize}
+	$\Rightarrow$ only 278 MFLOPS on 8 PES of CARY Y-MP \\
+	$\Rightarrow$ CRAY Y-MP (6 ns) $\rightarrow$ Theor. peak perf. = 333 Mflops (1 proc)
+	\end{frame}
+	\begin{frame}{Derivation of blocked algorithms}
+	$$
+		\begin{pmatrix}
+		A_{11} & a_{12} & A_{13} \\
+		. & a_{22} & a_{23} \\
+		. & . & A_{33}
+		\end{pmatrix}
+		=
+		\begin{pmatrix}
+		U_{11}^T & 0 & 0 \\
+		u_{12}^T & u_{22}^T & 0 \\
+		U_{13}^T & u_{23}^T & U_{33}^T
+		\end{pmatrix}
+		*
+		\begin{pmatrix}
+		U_{11} & u_{12}  & U_{13} \\
+		0 & u_{22} & u_{23} \\
+		0 & 0 & U_{33}
+		\end{pmatrix}
+	$$
+	Porównując macierze w drugim bloku kolumn mamy:
+	$
+	\begin{cases}
+		A_{12} = U_{11}^T * U_{12} \\
+		A_{22} = U_{12}^T * U_{12} + U_{22}^T * U_{22}
+	\end{cases}
+	$ \\ 
+	Jeżeli $U_{11}$ został już uprzednio wyliczony, to:
+	$$
+		U_{11}^T * U_{12} = A_{12} TRSM (Level 3 BLAS)
+	$$
+	$$
+		U_{22}^T * U_{22} = A_{22} - U_{12}^T * U_{12} SYRK(Level 2 BLAS)
+	$$
+	\end{frame}
```

Comment:
In the PDF, U12 and U22 are in some kind of boxes - did you ditch the boxes on purpose?